specs/eip4844/polynomial-commitments.md

```diff
@@ -414,6 +388,51 @@ def verify_kzg_proof_impl(commitment: KZGCommitment,
     ])
 ```
 
+#### `verify_kzg_proof_multi`
+
+```python
+def verify_kzg_proof_multi(commitments: Sequence[KZGCommitment],
+                           zs: Sequence[BLSFieldElement],
+                           ys: Sequence[BLSFieldElement],
+                           proofs: Sequence[KZGProof]) -> bool:
+    """
+    Verify multiple KZG proofs efficiently.
+    """
+
+    assert len(commitments) == len(zs) == len(ys) == len(proofs)
+
+    # Compute a random challenge. Note that it does not have to be computed from a hash,
```

Comment:
It feels a bit dirty doing here almost the same stuff we do in `compute_challenge()`. I wonder how we can abstract this functionality so that we don't have it separate.

Reply:
I think it's actually quite different since `compute_challenge` is now only for a single blob?